examples/AdvancedOperations/AddDynamicSearchAds.php

```diff
@@ -0,0 +1,347 @@
+use Google\Ads\GoogleAds\V1\Services\AdGroupOperation;
+use Google\Ads\GoogleAds\V1\Services\AdGroupAdOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignBudgetOperation;
+use Google\Ads\GoogleAds\V1\Services\CampaignOperation;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupAdsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateAdGroupCriteriaResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateCampaignBudgetsResponse;
+use Google\Ads\GoogleAds\V1\Services\MutateCampaignsResponse;
+use Google\Protobuf\Int64Value;
+use Google\Protobuf\StringValue;
+
+/**
+ * This example adds a new dynamic search ad (DSA) and webpage targeting criteria for the DSA.
+ */
+class AddDynamicSearchAds
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId
```

Comment:
We use this as "int" in other locations.
Could you please check if it should be string?
If so, we can then make a large change across all examples.

Reply:
You're right, it should be int as it needs to be without hyphens.